apis/infrastructure/v1beta1/bootstrapkubeconfig_types.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// DefaultClusterName is the cluster name used in the generated bootstrap-kubeconfig
+	DefaultClusterName = "default-cluster"
+
+	// DefaultContext is the context name used in the generated bootstrap-kubeconfig
+	DefaultContext = "default-context"
+
+	// DefaultNamespace is the namespace used in the generated bootstrap-kubeconfig
+	DefaultNamespace = "default"
+
+	// DefaultAuth is the auth in the generated bootstrap-kubeconfig
+	DefaultAuth = "default-auth"
+
+	// BootstrapTokenDescription is the bootstrap token description for the secret that is generated
+	BootstrapTokenDescription = "Token generated by cluster-api-provider-bringyourownhost"
+
+	// BootstrapTokenExtraGroups is the byoh group that has access to create CertificateSigningRequest
+	BootstrapTokenExtraGroups = "system:bootstrappers:byoh"
+)
+
+// BootstrapKubeconfigSpec defines the desired state of BootstrapKubeconfig
+type BootstrapKubeconfigSpec struct {
+	// APIServer is the address of the kubernetes cluster (https://hostname:port).
+	APIServer string `json:"apiserver"`
+
+	// InsecureSkipTLSVerify skips the validity check for the server's certificate. This will make your HTTPS connections insecure.
+	// +optional
+	// +kubebuilder:default=false
+	InsecureSkipTLSVerify bool `json:"insecure-skip-tls-verify,omitempty"`
+
+	// CertificateAuthorityData contains PEM-encoded certificate authority certificates.
+	CertificateAuthorityData string `json:"certificate-authority-data,omitempty"`
```

Comment:
CAData should be a required field?

Otherwise, we need to add an empty check [here](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/586/files#diff-77f5a1c2531cdb95f25926c95c7719b93f73d50c3c4e6622dd1cda9ba582762dR81).

Reply:
Yes, it is a required field. Good to have a webhook validating the correctness of this field. We also need the webhook to validate APIServer field to see if it is of the form `https://host:port`.
Will create a new issue for it.